SMLHelper/Patchers/PingTypePatcher.cs

```diff
@@ -0,0 +1,151 @@
+            if (PingManager.sCachedPingTypeStrings.valueToString.ContainsKey(pingType) == false)
+                PingManager.sCachedPingTypeStrings.valueToString.Add(pingType, name);
+
+            if (PingManager.sCachedPingTypeTranslationStrings.valueToString.ContainsKey(pingType) == false)
+                PingManager.sCachedPingTypeTranslationStrings.valueToString.Add(pingType, name);
+            
+            Logger.Log($"Successfully added PingType: '{name}' to Index: '{cache.Index}'", LogLevel.Debug);
+            return pingType;
+        }
+
+        internal static PingType? GetPingType(string pingName)
+        {
+            try
+            {
+                var ping = Enum.Parse(typeof(PingType), pingName);
+                return (PingType) ping;
+            }
+            catch (ArgumentException e)
+            {
+                return new PingType?();
+            }
+        }
+        
+        internal static void Patch(HarmonyInstance harmony)
+        {
+            IngameMenuHandler.Main.RegisterOneTimeUseOnSaveEvent(() => cacheManager.SaveCache());
+
+            harmony.Patch(AccessTools.Method(typeof(Enum), nameof(Enum.GetValues)),
+                postfix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Postfix_GetValues))));
+
+            harmony.Patch(AccessTools.Method(typeof(Enum), nameof(Enum.IsDefined)),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Prefix_IsDefined))));
+
+            harmony.Patch(AccessTools.Method(typeof(Enum), nameof(Enum.Parse), new Type[] { typeof(Type), typeof(string), typeof(bool) }),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Prefix_Parse))));
+
+            harmony.Patch(AccessTools.Method(typeof(PingType), nameof(PingType.ToString), new Type[] { }),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Prefix_ToString))));
+            
+            harmony.Patch(AccessTools.Method(typeof(SpriteManager), nameof(SpriteManager.Get), new Type[] { typeof(SpriteManager.Group), typeof(string) }),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Prefix_SpriteManager_Get))));
+
+            Logger.Log($"Added {cacheManager.ModdedKeysCount} PingTypes succesfully into the game.");
+            Logger.Log("PingTypePatcher is done.", LogLevel.Debug);
+        }
+
+        private static bool Prefix_SpriteManager_Get(SpriteManager.Group group, string name, ref Atlas.Sprite __result)
+        {
+            if (group == SpriteManager.Group.Pings)
+            {
+                var pingType = GetPingType(name);
+                if (pingType.HasValue && cachedSprites.ContainsKey(pingType.Value))
+                {
+                    __result = cachedSprites[pingType.Value];
+                    return false;
+                }
+            }
+
+            return true;
```

Comment:
We shouldn't prefix access to the SpriteManager.  
Instead, do how we have in the [SpritePatcher](https://github.com/SubnauticaModding/SMLHelper/blob/master/SMLHelper/Patchers/SpritePatcher.cs) and simply inject these sprites into the correct locations.

This may mean that the `cachedSprites` field goes away or is used only for working with the SpritePatcher.